lib/sequencer/src/execution/block_context_provider.rs

```diff
@@ -116,6 +118,28 @@ impl<Mempool: L2TransactionPool> BlockContextProvider<Mempool> {
 
                 let timestamp = (millis_since_epoch() / 1000) as u64;
 
+                // Check if we peeked an upgrade transaction info.
+                let mut force_preimages = Vec::new();
```

Comment:
nit - let's maybe make it immutable? `let force_preimages = if {} else {vec![]}` . This way it's immediately clear that `force_preimages`  is only populated for upgrades and not mutated further down for any other reason